app/controllers/users_controller.rb

```diff
@@ -1,4 +1,6 @@
 class UsersController < ApplicationController
+  before_action :require_user, only: :show
+  
   def new
 
   end
```

Comment:
This method is where the API key needs to get stored also for our client's queries!

Reply:
The method "require_user"? This lives in the application controller. I would be worried about storing the API key here since it can be exposed on Github. Perhaps I'm misunderstanding your comment?